app/services/bookings/gitis/store/write_only_cache.rb

```diff
@@ -0,0 +1,48 @@
+module Bookings
+  module Gitis
+    module Store
+      class WriteOnlyCache
+        attr_reader :store, :cache, :namespace, :ttl, :version
+        delegate :fetch, to: :store
+
+        def initialize(store, cache, namespace: nil, ttl: 1.hour, version: 'v1')
+          @store = store
+          @cache = cache
+          @namespace = namespace
+          @ttl = ttl
+          @version = version
+        end
+
+        def cache_key_for_entity(entity)
+          [namespace, entity.cache_key, version].compact.join('/')
+        end
+
+        def find(entity_type, uuids, **options)
+          find_and_write_to_cache(entity_type, uuids, **options)
+        end
+
+        def write(entity)
+          store.write(entity).tap do
+            cache.delete cache_key_for_entity entity
+          end
+        end
+
+      private
+
+        def entities_to_cache(entities)
+          Array.wrap(entities).each.with_object({}) do |entity, cache_set|
+            cache_set[cache_key_for_entity(entity)] = entity.to_cache
+          end
+        end
+
+        def find_and_write_to_cache(entity_type, uuids, **options)
+          store.find(entity_type, uuids, **options).tap do |entities|
+            if options.empty?
```

Comment:
this is never empty - if `includes:` is set to nil then this still ends up set to `{includes:nil}` which means the cache is never written to

Reply:
Do'h had tested with keys set from an earlier version which passed and empty options hash in from Gitis::CRM instead of `{includes: nil}` always being set - easy fix